nexus/transport.go

```diff
@@ -0,0 +1,707 @@
+		//     [OperationHandle] will be returned as StartOperationResponse.Pending, which can be used to perform actions
+		//     such as getting its result.
+		//
+		//  4. There was an error making the call. The returned response will be nil and the error will be a
+		//     [HandlerError].
+		//
+		//  5. Any other error.
+		//
+		// NOTE: Experimental
+		StartOperation(
+			ctx context.Context,
+			service string,
+			operation string,
+			input any,
+			options StartOperationOptions,
+		) (*StartOperationResponse[*LazyValue], error)
+
+		// GetOperationInfo returns information about a specific operation. Should only be called through an
+		// [OperationHandle].
+		//
+		// NOTE: Experimental
+		GetOperationInfo(
+			ctx context.Context,
+			service string,
+			operation string,
+			token string,
+			options GetOperationInfoOptions,
+		) (*GetOperationInfoResponse, error)
+
+		// GetOperationResult gets the result of an operation, issuing a network request to the service handler.
+		//
+		// By default, GetResult returns (nil, [ErrOperationStillRunning]) immediately after issuing a call if the
+		// operation has not yet completed.
+		//
+		// Callers may set GetOperationResultOptions.Wait to a value greater than 0 to alter this behavior, causing
+		// the transport to long poll for the result issuing one or more requests until the provided wait period
+		// exceeds, in which case (nil, [ErrOperationStillRunning]) is returned.
+		//
+		// The wait time is capped to the deadline of the provided context. Make sure to handle both context deadline
+		// errors and [ErrOperationStillRunning].
+		//
+		// Note that the wait period is enforced by the server and may not be respected if the server is misbehaving.
+		// Set the context deadline to the max allowed wait period to ensure this call returns in a timely fashion.
+		//
+		// ⚠️ If a [LazyValue] is returned (as indicated by T), it must be consumed to free up the underlying connection.
+		//
+		// Should only be called through an [OperationHandle].
+		//
+		// NOTE: Experimental
+		GetOperationResult(
+			ctx context.Context,
+			service string,
+			operation string,
+			token string,
+			options GetOperationResultOptions,
+		) (*GetOperationResultResponse[*LazyValue], error)
+
+		// CancelOperation requests to cancel an asynchronous operation.
+		//
+		// Cancelation is asynchronous and may be not be respected by the operation's implementation.
+		//
+		// Should only be called through an [OperationHandle].
+		//
+		// NOTE: Experimental
+		CancelOperation(
+			ctx context.Context,
+			service string,
+			operation string,
+			token string,
+			options CancelOperationOptions,
+		) (*CancelOperationResponse, error)
+	}
+
+	// StartOperationResponse is the response to Transport.StartOperation calls. One and only one of Complete or
+	// Pending will be populated.
+	//
+	// NOTE: Experimental
+	StartOperationResponse[T any] struct {
```

Comment:
IMO all of this stuff could be prefixed with `Transport` and make it transport only. A clean separation between the transport impl and the actual client impl has value IMO compared to reusing transport types in non-transport situations or cluttering the API with transport-only types unqualified.